### PR TITLE
fix(Workspaces): use UUID scalar for workspace

### PR DESCRIPTION
**Pull request by cheikhgwane**
Fix for https://github.com/orgs/BLSQ/projects/4/views/2?filterQuery=label%3Abug&pane=issue&itemId=18282101.

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- Change 1
- Change 2

## How/what to test

A few words about what needs to be tested, along with specific testing instructions if needed.

## Screenshots / screencast

If relevant, please add some screenshots or a short video.

### CODE REVIEW

**Comment by cheikhgwane** [comment]
> Would be nice to change all query fields that use a `id: String!` by this `UUID` type.

For the workspaces schema only ?

**Comment by qgerome** [comment]
No, all of the ids.
Quentin Gérôme
+32 471 490 190

**Comment by qgerome** [comment]
You can also update the mutations to take this UUID instead of the String when it makes sense.

File: hexa/workspaces/schema.py [hunk 1]
```diff
@@ -21,9 +28,19 @@
...
+uuid_scalar = ScalarType("UUID")
+
+
+@uuid_scalar.value_parser
+def parse_uuid_value(value):
+    try:
+        UUID(value, version=4)
+        return str(value).upper()
+    except (ValueError, TypeError):
+        raise ValueError(f'"{value}" is not a valid input')
```
Comment (qgerome) [inline hexa/workspaces/schema.py:42]:
```suggestion
        raise ValueError(f'"{value}" is not a valid uuid')
```

**Review by qgerome — CHANGES_REQUESTED** [review]

**Pull request merged**